docs/fastdds/dds_layer/subscriber/dataReader/readingData.rst

```diff
@@ -114,7 +114,15 @@ Accessing data with a waiting thread
 Instead of relying on the Listener to try and get new data values,
 the application can also dedicate a thread to wait until any new data is available on the
 DataReader.
-This can be done with the :func:`wait_for_unread_message` member function,
+This can be done using a wait-set to wait for a change on the `DataAvailable` status.
+
+.. literalinclude:: /../code/DDSCodeTester.cpp
+   :language: c++
+   :start-after: //DDS_DATAREADER_READ_WAITSET
```

Comment:
The labels are treated as regex, so the original `//DDS_DATAREADER_READ_WAIT` (used down below) gets mixed with this `//DDS_DATAREADER_READ_WAITSET` and the same block of code is used in the output.

Change either label to avoid this . For example, use `DDS_DATAREADER_WAIT_FOR_UNREAD` on the old block.

Reply:
Done on 130ae2b